packages/web3-eth-accounts/src/account.ts

```diff
@@ -240,7 +241,7 @@ export const sign = (data: string, privateKey: Bytes): SignResult => {
  * }
  * ```
  */
-export const signTransaction = async (
+export const signTransactionWithPrivateKey = async (
```

Comment:
same here: This will add extra step for users to migrate to 4.x, and we should make it least effort for end users instead of additional steps. so I think this change should be reverted.